packages/api/src/helpers/extensionMessaging.ts

```diff
@@ -62,3 +62,30 @@ export const sendMessageToContentScript = (msg: APIMessages): Promise<any> => {
     window.addEventListener('message', messageListener, false);
   });
 };
+
+/**
+ * Serializes the message to be sent to the content script.
+ * Fields that are objects are stringified:
+ * - memos
+ * - amount
+ * - flags
+ *
+ * @param msg
+ * @returns The message with each object field stringified.
+ */
+const serializeMessage = (msg: APIMessages): any => {
+  const modifiedMsg: any = { ...msg };
+  if ('payload' in modifiedMsg && 'memos' in modifiedMsg.payload && modifiedMsg.payload.memos) {
+    modifiedMsg.payload.memos = JSON.stringify(modifiedMsg.payload.memos);
+  }
+
+  if ('payload' in modifiedMsg && 'amount' in modifiedMsg.payload && typeof modifiedMsg.payload.amount === 'object') {
+    modifiedMsg.payload.amount = JSON.stringify(modifiedMsg.payload.amount);
+  }
```

Comment:
You could write it like this (which would be easier to read):

if (typeof modifiedMsg.payload?.amount === 'object') {
  modifiedMsg.payload.amount = JSON.stringify(modifiedMsg.payload.amount);
}

Would you like to stringify if amount===null

In your code or the code I propose it would stringify the null value.